### PR TITLE
Feature/hmi 03 04 2020

### PR DESCRIPTION
**Pull request by gustavosinbandera1**
This PR is in order to handle a input button events  to control the turpial device, for now the code is capable of handling three different events on the same button, click, double click and long click, and based on that turn RGB leds on/off to test the feature, for now I am not sure about how to handle radio device to enable or disable it, I'll need to check the schematic tomorrow, the second approach could be enable and disable AP/STA wifi services, 
inside main file, there is a user button object to build any new input button to assign new functionalities, just  any need to connect a button b/w desired GPIO in constructor or default user button defined inside defaults.h (GPIO_NUM_21)

### CODE REVIEW

**Comment by gustavosinbandera1** [comment]
this branch could be merged, in the actual  implementation the code is able to handle any action over different button events, I have fixed many bad names for variables as well

File: components/HMI/src/HumanInterface.cpp [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+void Button::attachLongClick(callbackFunction newFunction)
+{
+    _longClickFunc = newFunction;
+} // attachClick
+
+
+/**
+ * @brief reset the state machine, the proccess function was finished
+ * 
+ */
```
Comment (luisan00) [inline components/HMI/src/HumanInterface.cpp]:
```suggestion
/**
 * @brief reset the state machine, the process function was finished
 * 
 */
```

File: components/HMI/src/UserButton.cpp [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    io_conf.pull_up_en = GPIO_PULLUP_DISABLE;
+    gpio_config(&io_conf);
+
+    //interrupt of rising edge
+    io_conf.intr_type = GPIO_INTR_POSEDGE;
+    io_conf.pin_bit_mask = (1ULL << btn->_gpio_btn); //GPIO_INPUT_PIN_SEL;
+    io_conf.mode = GPIO_MODE_INPUT;
+    io_conf.pull_up_en = GPIO_PULLUP_ENABLE;
+    gpio_config(&io_conf);
+    //change gpio intrrupt type
```
Comment (luisan00) [inline components/HMI/src/UserButton.cpp]:
```suggestion
    //change gpio interrupt type
```

File: components/HMI/src/UserButton.cpp [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+}
+
+/**
+ * @brief Interrupt class is implementing run virtual method from Task class
+ * to has ineer loop task, also we are using the same loop to handle state machine
+ * to debounce input button.
+ * The task first blocks waiting for a communications event inside firs do-while loop. 
+ * The block time is relatively short. 
+ * the second do-while block waiting for some states to decode the button input.
+ * The methid run is getting and instance of Button class as a parameter
```
Comment (luisan00) [inline components/HMI/src/UserButton.cpp]:
```suggestion
 * The method run is getting and instance of Button class as a parameter 
```

**Comment by gustavosinbandera1** [comment]
the spelling errors were fixed

**Pull request merged**